rustls/src/error.rs

```diff
@@ -97,6 +97,9 @@ pub enum Error {
     /// or too large.
     BadMaxFragmentSize,
 
+    /// Specific failure cases from [`crate::crypto::signer::CertifiedKey::keys_match`].
+    InconsistentKeys(InconsistentKeys),
```

Comment:
I would be tempted to combine this commit with the one that follows that adds `keys_match`. I think the implementation + the error it can produce being together in one unit makes the most sense.